utils/s2n_random.c

```diff
@@ -60,14 +56,19 @@
 
 static int entropy_fd = UNINITIALIZED_ENTROPY_FD;
 
-static __thread struct s2n_drbg per_thread_private_drbg = {0};
-static __thread struct s2n_drbg per_thread_public_drbg = {0};
-
-static void *zeroed_when_forked_page = NULL;
-static int zero = 0;
+struct s2n_rand_state {
+    uint64_t cached_fork_generation_number;
+    struct s2n_drbg public_drbg;
+    struct s2n_drbg private_drbg;
+    bool drbgs_initialised;
+};
 
-static __thread void *zero_if_forked_ptr = &zero;
-#define zero_if_forked (* (int *) zero_if_forked_ptr)
+static __thread struct s2n_rand_state s2n_per_thread_rand_state = {
+    .cached_fork_generation_number = 0,
+    .public_drbg = {0},
+    .private_drbg = {0},
+    .drbgs_initialised = false
+};
```

Comment:
Nit: drbgs_ready, drbgs_inited, or drbgs_initiali**z**ed 

Sorry for being pedantic about this, it just seems a little silly when we spell it two different ways in the same file / repo.